docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,652 @@
+reviewed, audited, and have billions of transactions under their belt, boasting
+super high stability. This approach can make your code smaller and more
+reliable.
+
+In other platforms, an audit usually involves auditing everything, including the
+dependencies. However, in Gno, we can expect that over time, contracts will
+become smaller, more powerful, and partially audited by default, thanks to this
+enforced open-source system.
+
+So, while you can still adhere to the original philosophy of minimizing
+dependencies, ultimately, try to use and write super stable, simple, tested,
+focused `p/` small libraries. This approach can lead to more reliable,
+efficient, and trustworthy Gno contracts.
+
+```go
+import (
+    "gno.land/p/finance/tokens"
+    "gno.land/p/finance/exchange"
+    "gno.land/p/finance/wallet"
+    "gno.land/p/utils/permissions"
+)
+
+var (
+    myWallet wallet.Wallet
+    myToken tokens.Token
+    myExchange exchange.Exchange
+)
+
+func init() {
+    myWallet = wallet.NewWallet()
+    myToken = tokens.NewToken("MyToken", "MTK")
+    myExchange = exchange.NewExchange(myToken)
+}
+
+func BuyTokens(amount int) {
+    caller := permissions.GetCaller()
+    permissions.CheckPermission(caller, "buy")
+    myWallet.Debit(caller, amount)
+    myExchange.Buy(caller, amount)
+}
+
+func SellTokens(amount int) {
+    caller := permissions.GetCaller()
+    permissions.CheckPermission(caller, "sell")
+    myWallet.Credit(caller, amount)
+    myExchange.Sell(caller, amount)
+}
+```
+
+##  When Gno takes Go practices to the next level
+
+### Documentation is for users
+
+One of the well-known proverbs in Go is: ["Documentation is for
+users"](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=1147s), as stated by Rob
+Pike. In Go, documentation is for users, but users are often developers. In Gno,
+documentation is for users, but users can be another developer or the end users.
+
+In Go, we usually have well-written documentation for other developers to
+maintain and use our code as a library. Then, we often have another layer of
+documentation on our API, sometimes with OpenAPI Specs, Protobuf, or even user
+documentation.
+
+In Gno, the focus shifts towards writing documentation for the end user. You can
+even consider that the main reader is an end user, who is not so interested in
+technical details, but mostly interested in how and why they should use a
+particular endpoint. Comments will be used for code source reading, but also to
+generate documentation and even for smart wallets that need to understand what
+to do.
+
+Inline comments have the same goal: to guide users (developers or end users)
+through the code. While comments are still important for maintainability, their
+main purpose in Gno is for discoverability. This shift towards user-centric
+documentation reflects the broader shift in Gno towards making code more
+accessible and understandable for all users, not just developers.
+
+TODO: `func ExampleXXX`.
```

Comment:
Leftover?

Reply:
I'm happy to merge with this TODO. Please feel free to propose an implementation now.

Reply:
Should be fine for now, there is a list at the bottom of the doc to resolve dangling TODOs